tests/dag_processing/test_processor.py

```diff
@@ -404,7 +404,7 @@ def test_process_file_should_failure_callback(self):
 
     @conf_vars({("core", "dagbag_import_error_tracebacks"): "False"})
     def test_add_unparseable_file_before_sched_start_creates_import_error(self, tmpdir):
-        unparseable_filename = tmpdir / TEMP_DAG_FILENAME
+        unparseable_filename = os.path.join(tmpdir, TEMP_DAG_FILENAME)
```

Comment:
These changes are required because the previous `tmpdir / "child_path"` syntax would return a PathLib object rather than a string, and the `_process_file` function expects a string. 

Providing the wrong type was causing some of the tests to fail due SQLAlchemy errors.

Using `os.path.join` will explicitly return a string and thus fixes this issue.

Reply:
`tmpdir` is not a `pathlib.Path` but a `py._path.LocalPath`, which is a path object implementation in `pytest` (actually from a package `py` that backs most of `pytest`) that pre-dates the stdlib `pathlib`.

But you fix is correct—`py._path.LocalPath` implements the new [fspath protocol](https://docs.python.org/3/library/os.html#os.PathLike) introduced in Python 3.6 (PEP 519), so it can be consumed by `os.path.join` and coerced into a str.